### PR TITLE
Initial implementation of Stack

### PR DESCRIPTION
**Pull request by Den4eg007**

### CODE REVIEW

File: src/main/java/core/stack/MateStack.java [hunk 1]
```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
```
Comment (KseniaObertii) [inline src/main/java/core/stack/MateStack.java:6]:
```suggestion
    private Node<T> top;
```

File: src/main/java/core/stack/MateStack.java [hunk 2]
```diff
@@ -1,19 +1,50 @@
...
+    private int size;
 
+    public void push(T value) {
+        if (tail == null) {
+            tail = new Node<>(value, null);
+        } else {
+            tail = new Node<>(value, tail);
+        }
```
Comment (KseniaObertii) [inline src/main/java/core/stack/MateStack.java:14]: top is already null, we don't need first check
```suggestion
            tail = new Node<>(value, tail);
```

File: src/main/java/core/stack/MateStack.java [hunk 3]
```diff
@@ -1,19 +1,50 @@
...
     public T peek() {
-        return null;
+        checkSize();
+        return tail.value;
     }
 
     public T pop() {
-        return null;
+        checkSize();
+        T element = tail == null ? null : tail.value;
```
Comment (KseniaObertii) [inline src/main/java/core/stack/MateStack.java:25]:
```suggestion
        T element = peek()
```

File: src/main/java/core/stack/MateStack.java [hunk 4]
```diff
@@ -1,19 +1,50 @@
...
+        tail = tail == null ? null : tail.prev;
```
Comment (KseniaObertii) [inline src/main/java/core/stack/MateStack.java:26]: we don't need this check
```suggestion
        tail = tail.prev;
```

**Review by KseniaObertii — CHANGES_REQUESTED** [review]

**Review by KseniaObertii — APPROVED** [review]
That's it, man

**Review by Max-Pochepets — APPROVED** [review]

**Pull request closed without merging**